src/js/base/js-numbers.js

```diff
@@ -191,7 +191,7 @@ define("pyret-base/js/js-numbers", function() {
     var match = s.match(scientificPattern), mantissaChunks, exponent;
     if (match) {
       mantissaChunks = match[1].match(/^([^.]*)(.*)$/);
-      exponent = Number(match[2]);
+      exponent = Number(match[2]); //possible, but not probable, fixnum overflow!
```

Comment:
This is worth handling.  Would it be easier to try using `fromString` here (which already can handle big numbers) and then checking the size of the result?

Reply:
If the exponent is not a fixnum, I can make it into a bigint and create intermediate number strings accordingly to construct the larger bigint. This can be done (and errbacks aren't needed, because there is no algorithmically determinable overflow anywhere in the process) and I'll push soon. 

However, the reason I said it was improbable is because Pyret chokes on even moderately large _fixnums_ in the exponent position. So we never get to the point of being able to test if a bigint exponent works.

Try something as "small" as `9e999999` (six 9s in the exponent). The entire IDE (and web page) collapses under the weight of the underlying computation.

Reply:
Granted; there are lots of numbers we can produce that collapse the browser due to their computational magnitude.  That *does not mean* that "improbable" is an acceptable failure mode.  If Pyret crashes the browser in its entirety, that's a better user experience than showing the user "errors prevented Pyret from showing the full message" -- the former says we're trying to do our best and the browser can't keep up, while the latter says we screwed up.  If a user types `1e999999999999999999999999999................999`, and we crash because `Number("999............999999999999999999")` screws up in some way (e.g. by returning `Infinity`), then that's not an ok response.  We need to at least create the number representing the exponent, and choose to raise a domain exception instead of letting the page crash.

As for errbacks not being needed: I'm not sure what you're arguing for or against here -- errbacks are needed in any function that transitively may call a function that may throw an error.  And for methods on our number hierarchy, if any of them might use errbacks, then all of them must take errbacks, because that's a part of their signature.  If _this particular_ helper function doesn't need errbacks right now, but also has JS-level failure modes, then maybe this function needs to be redesigned to not crash at the JS level, and that may entail adding errbacks.

Reply:
I actually fixed the issue in my follow-up push (18f7e6d) to my PR. Exponents that are not fixnum are technically allowed. I am, however, unable to devise a meaningful test of this fix, at least on my computer, because of page collapse. Screwing up because of an `Infinity` is definitely not countenanced anymore.

I can certainly add `errbacks` to `expandExponent` (and that would entail adding it to `makeBignum` too), but right now, none of the functions called within the dynamic extent of either of these functions invokes `errbacks.throw*`.

I would have needed to add `errbacks` if I was having `expandExponent` raise error on a non-fixnum exponent, but the current fix addresses such exponents fully, and will give an answer _if your computer permits_. Please review that it is satisfactory: the change is localized to the body of `expandExponent`, with one helper function called `zfillBignum`

Reply:
P.S. When I say non-fixnum exponents are now allowed, I meant of course exponents that are strings of digits only, not embedded exponent notation. We do not (intentionally?) address numbers such as `1e999e999` (i.e., with multiple `e`s).